test/e2e/storage/subpath.go

```diff
@@ -671,131 +813,6 @@ func podContainerExec(pod *v1.Pod, containerIndex int, bashExec string) (string,
-		source: &v1.VolumeSource{
-			HostPath: &v1.HostPathVolumeSource{
-				Path: "/tmp",
-			},
-		},
-	}
-}
-
-func (s *hostpathSource) getReadOnlyVolumeSpec() *v1.VolumeSource {
-	return nil
-}
-
-func (s *hostpathSource) cleanupVolume(f *framework.Framework) {
-}
-
-type hostpathSymlinkSource struct {
-}
-
-func initHostpathSymlink() volSource {
-	return &hostpathSymlinkSource{}
-}
-
-func (s *hostpathSymlinkSource) createVolume(f *framework.Framework) volInfo {
-	nodes := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
-	Expect(len(nodes.Items)).NotTo(BeZero(), "No available nodes for scheduling")
-
-	node0 := &nodes.Items[0]
-	sourcePath := fmt.Sprintf("/tmp/%v", f.Namespace.Name)
-	targetPath := fmt.Sprintf("/tmp/%v-link", f.Namespace.Name)
-	cmd := fmt.Sprintf("mkdir %v -m 777 && ln -s %v %v", sourcePath, sourcePath, targetPath)
-	privileged := true
-
-	// Launch pod to initialize hostpath directory and symlink
-	pod := &v1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: fmt.Sprintf("hostpath-symlink-prep-%s", f.Namespace.Name),
-		},
-		Spec: v1.PodSpec{
-			Containers: []v1.Container{
-				{
-					Name:    fmt.Sprintf("init-volume-%s", f.Namespace.Name),
-					Image:   "busybox",
-					Command: []string{"/bin/sh", "-ec", cmd},
-					VolumeMounts: []v1.VolumeMount{
-						{
-							Name:      volumeName,
-							MountPath: "/tmp",
-						},
-					},
-					SecurityContext: &v1.SecurityContext{
-						Privileged: &privileged,
-					},
-				},
-			},
-			RestartPolicy: v1.RestartPolicyNever,
-			Volumes: []v1.Volume{
-				{
-					Name: volumeName,
-					VolumeSource: v1.VolumeSource{
-						HostPath: &v1.HostPathVolumeSource{
-							Path: "/tmp",
-						},
-					},
-				},
-			},
-			NodeName: node0.Name,
-		},
-	}
-	pod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(pod)
-	Expect(err).ToNot(HaveOccurred(), "while creating hostpath init pod")
-
-	err = framework.WaitForPodSuccessInNamespace(f.ClientSet, pod.Name, pod.Namespace)
-	Expect(err).ToNot(HaveOccurred(), "while waiting for hostpath init pod to succeed")
-
-	err = framework.DeletePodWithWait(f, f.ClientSet, pod)
-	Expect(err).ToNot(HaveOccurred(), "while deleting hostpath init pod")
-
-	return volInfo{
-		source: &v1.VolumeSource{
-			HostPath: &v1.HostPathVolumeSource{
-				Path: targetPath,
-			},
-		},
-		node: node0.Name,
-	}
-}
-
-func (s *hostpathSymlinkSource) getReadOnlyVolumeSpec() *v1.VolumeSource {
-	return nil
-}
-
-func (s *hostpathSymlinkSource) cleanupVolume(f *framework.Framework) {
-}
-
-type emptydirSource struct {
-}
-
-func initEmptydir() volSource {
-	return &emptydirSource{}
-}
-
-func (s *emptydirSource) createVolume(f *framework.Framework) volInfo {
-	return volInfo{
-		source: &v1.VolumeSource{
-			EmptyDir: &v1.EmptyDirVolumeSource{},
-		},
-	}
-}
-
-func (s *emptydirSource) getReadOnlyVolumeSpec() *v1.VolumeSource {
-	return nil
-}
-
-func (s *emptydirSource) cleanupVolume(f *framework.Framework) {
-}
-
 type gcepdPVCSource struct {
```

Comment:
Please move this GCE PD to non_csi_volumes.go too, we'll add cinder and AWS later.

Reply:
I will try it. However, it seems that current initGCEPDPartition is always skipped due to the fact that it requires manual operations. So, I'm not sure that I can test my code properly.

Reply:
that's fine, I can help validate gce pd as needed